### DISCUSSION TITLE
Everything renders correctly but background colors are not rendering at all

### DISCUSSION THREAD

**Discussion by NimbusFox** [mikke89/RmlUi#444] [Q&A] [answered]
[My renderer](https://github.com/NimbusFox/raylib-RmlUi/blob/main/src/raylibRenderInterface.cpp#L7)

I am using an easy to use OpenGL wrapper called Raylib. I am in the process of implementing RmlUi but I can't seem to wrap my head around why background colors are not being rendered. Any advice would be very welcome.

The examples work correctly. But when I run it in my own code and renderer no color backgrounds work

My renderer
![Render with no background colors](https://user-images.githubusercontent.com/7228476/230226116-38f0619b-4e10-443a-b141-6b71eccf2761.png)
```html
<rml>
    <head>
        <title>Example</title>
        <style>
            body {
                position: absolute;
                top: 50px;
                left: 50px;
                width: 500px;
                height: 500px;
                background-color: #ccc;
            }

            div {
                display: block;
                height: 150px;
                width: 200px;
                background-color: #f00;
            }
        </style>
    </head>
    <body>
    <div/>
    </body>
</rml>
```

Demo Sample
![image](https://user-images.githubusercontent.com/7228476/230226224-4bd49b49-fd80-4a31-9a7e-b08bc1b1c0f6.png)

**Comment by mikke89** [comment]
Hey. My guess is that vertex colors are not applied correctly. Check your renderer and shaders, and ensure that the vertex colors are multiplied with any texture color. The samples use texture-based backgrounds for the most part, so this issue is not immediately apparent there.

**Comment by NimbusFox** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/7228476/230242304-c5373c47-9334-4597-97fd-b67db4e37e6f.png)
Spoke with raylib community. The issue was if there is no texture supplied my code would skip over it. For future reference for anyone. If there is no texture supplied. Return a 2x2 white square as it's texture
